streaming/src/main/scala/io/gearpump/streaming/appmaster/JarScheduler.scala

```diff
@@ -32,43 +36,105 @@ import io.gearpump.util.Graph
  * parts, with each part use its own jar file.
  *
  */
-class JarScheduler(appId : Int, appName: String, config: Config) {
-  private var taskSchedulers = Map.empty[AppJar, TaskScheduler]
-
-  def setDag(dag: DAG): Unit = {
-    val processors = dag.processors.values.groupBy(_.jar)
-    taskSchedulers = processors.map { jarAndProcessors =>
-      val (jar, processors) = jarAndProcessors
-      //Construct the sub DAG
-      val graph = Graph.empty[ProcessorDescription, PartitionerDescription]
-      processors.foreach(graph.addVertex)
-      val subDag = DAG(graph)
-      val taskScheduler = taskSchedulers.getOrElse(jar, new TaskSchedulerImpl(appId, appName, config))
-      taskScheduler.setDAG(subDag)
-      jar -> taskScheduler
+class JarScheduler(appId : Int, appName: String, config: Config, factory: ActorRefFactory) {
+  private val actor: ActorRef = factory.actorOf(Props(new JarSchedulerImpl(appId, appName, config)))
```

Comment:
It is not OK. there are other calls to jarScheduler.xxmethod in TaskManager. Since map execute in a future, we cannot gurantee setDag is executed before  .xxmethod